### DISCUSSION TITLE
Joomla 4 Login JWT

### DISCUSSION THREAD

**Discussion by aminweb2** [joomla/joomla-cms#35643] [Q&A]
Hello Joomlars!
in Joomla 4, how can I login authentication users in Ionic Framework or Flutter or ...? 
I know for get users list : I can use this command :Get: {{base_path}}/api/index.php/v1/users  + Token bearer

but for login authentication users , What command should I use?

**Comment by PhilETaylor** [comment]
The token **is** the authentication...

**Reply by aminweb2** [reply]
My question is: Does Joomla! 4 Have JWT for use in login page on mobile app framework like ionic?

**Reply by PhilETaylor** [reply]
No.

**Reply by aminweb2** [reply]
So how do I create it?

**Reply by PhilETaylor** [reply]
You dont. The token you use is a per user token, and has the rights of that user only. Joomla doesnt provide a JWT integration for "login"

**Reply by aminweb2** [reply]
Is there no way to create a login page in a mobile app for Joomla?

**Reply by PhilETaylor** [reply]
No.

https://magazine.joomla.org/all-issues/august-2020/joomla-web-services-api-101-tokens,-testing-and-a-taste-test
https://docs.joomla.org/J4.x:Joomla_Core_APIs

**Reply by aminweb2** [reply]
so how created this app mobile for joomla?!
https://www.cmsjunkie.com/business-directory-android-mobile-app

**Reply by PhilETaylor** [reply]
No idea. But that is also not interfacing with Joomla 4 Web Services.

**Reply by aminweb2** [reply]
I'm just looking for a way to create this. It does not matter where the source is.

**Reply by PhilETaylor** [reply]
There is no built in way to check the username/password of a user using Joomla 4 API Webservices. You will need to design your own way to do this, or use the token authentication that is built into Joomla 4.

**Reply by aminweb2** [reply]
can you explain more? or sample code

**Reply by dgrammatiko** [reply]
@aminweb2 my request for JWT was gloriously ignored https://github.com/joomla/joomla-cms/issues/27614#issuecomment-578792358 . I mean I get that any frontenter will want to just use a jwt token in their SPA/MPA app as this is the industry standard what I don't get is why this is ignored...

**Reply by aminweb2** [reply]
@dgrammatiko why ignored !!?? Do you have a solution to this problem?

**Comment by wilsonge** [comment]
The token is the only authentication offered out of the box. Web Service Auth is plugin driven so people can create their own plugins to do JWT's or OAuth or anything similar
